### PR TITLE
Add missing parameter to Set-AzureRmPolicyDefinition

### PR DESCRIPTION
**Pull request by mentat9**
## Description

Bug fix: add missing parameter to Set-AzureRmPolicyDefinition

## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [X] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [X] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Resources/Commands.Resources/help/Set-AzureRmPolicyDefinition.md [hunk 1]
```diff
@@ -82,7 +90,7 @@ Accept wildcard characters: False
 The credentials, account, tenant, and subscription used for communication with azure
 
 ```yaml
-Type: Microsoft.Azure.Commands.Common.Authentication.Abstractions.IAzureContextContainer
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources/help/Set-AzureRmPolicyDefinition.md]: @mentat9 would you mind running the `Update-MarkdownHelpModule` command again with the `-UseFullTypeName` parameter to preserve the fully qualified name in the types?

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/PolicyTests.ps1 [hunk 2]
```diff
@@ -66,6 +66,154 @@ function Test-PolicyDefinitionCRUD
     Assert-AreEqual True $remove
 }
 
+<#
+.SYNOPSIS
+Tests Policy definition CRUD operations
+#>
+function Test-PolicyDefinitionMode
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/PolicyTests.ps1]: 🎉 🎉

File: src/ResourceManager/Resources/Commands.Resources.Test/Commands.Resources.Test.csproj [hunk 3]
```diff
@@ -294,6 +294,7 @@
     <None Include="SessionRecords\Microsoft.Azure.Commands.Resources.Test.ScenarioTests.PolicyTests\TestPolicyAssignmentIdentity.json">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
+    <None Include="SessionRecords\Microsoft.Azure.Commands.Resources.Test.ScenarioTests.PolicyTests\TestPolicyDefinitionMode.json" />
```
[the remark this answers is not in the document]
Comment (mentat9) [reply src/ResourceManager/Resources/Commands.Resources.Test/Commands.Resources.Test.csproj]: Right, thanks. I always forget about that particular pitfall...

File: src/ResourceManager/Resources/Commands.Resources/ChangeLog.md [hunk 4]
```diff
@@ -18,6 +18,7 @@
         - Additional information about change #1
 -->
 ## Current Release
+* Add missing -Mode parameter to Set-AzureRmPolicyDefinition
```
[the remark this answers is not in the document]
Comment (cormacpayne) [reply src/ResourceManager/Resources/Commands.Resources/ChangeLog.md]: Yup! Looks good

**Pull request merged**